src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocCache4.java

```diff
@@ -0,0 +1,15 @@
+/*
+SummaryJavadoc
+violateExecutionOnNonTightHtml = (default)false
+forbiddenSummaryFragments = (default)^$
+period = (default).
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.abstractjavadoc;
+
+// violation below 'First sentence of Javadoc is missing an ending period.'
+/** Summary. */ /** some text*/
+public class InputAbstractJavadocCache4 {
```

Comment:
This is expected behavior. Good.
But need to look closer to code explain why.
As I think we will not parse second comment as it is in cache after first comment is parsed

Reply:
I understand your point, I'm willing to give it a try and understand what's going on. 
I still have other stuff to work on though.

Do you think it is worth the time investment ? Or maybe the risk to apply this update is too high for the benefit ?

Reply:
Reason it works is because `JavadocUtil.isJavadocComment` is smart and do not consider first in line block comment a javadoc.

https://github.com/checkstyle/checkstyle/blob/a1ea36d1db4d78527e7ae2c5d0501c083fa3c586/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java#L318-L319


https://github.com/checkstyle/checkstyle/blob/a1ea36d1db4d78527e7ae2c5d0501c083fa3c586/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtil.java#L160-L162


https://github.com/checkstyle/checkstyle/blob/a1ea36d1db4d78527e7ae2c5d0501c083fa3c586/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtil.java#L340-L356

Reply:
@romani okay thanks for the heads up. 
You found the reason for the behavior so It's good news for the current matter. 

I was currently going down the inner workings of the TreeWalker and noticed something similar. 

For future reference I would like point out that the JavaLanguageLexer#reportBlockComment shows the same behavior and caches as Javadoc only the last Javadoc block,
specifically this piece of code:

<img width="890" alt="image" src="https://github.com/user-attachments/assets/84893b1a-299b-4b81-bdc8-f7b2b09e6547">
So basically since the map key is the line number, the last comment is overridden by the next comment found on the same line.

Anyways, I thought it was worth pointing out.